### PR TITLE
[WIP] Test runner changes

### PR DESCRIPTION
**Pull request by urso**

### CODE REVIEW

**Comment by andrewkroh** [comment]
```
OSError: [Errno 13] Permission denied: '/tmp/python-env/lib/python2.7/site-packages/urllib3-1.21.1.dist-info/DESCRIPTION.rst'
scripts/Makefile:203: recipe for target 'python-env' failed
```

It took me a while to figure out the cause, but I believe the cause is that the [Dockerfile](https://github.com/elastic/beats/blob/master/libbeat/Dockerfile#L13-L18) pre-builds the virtual environment as root.

**Comment by ruflin** [comment]
It would be great if we get this working and can get rid of the cleanup.

**Comment by ruflin** [comment]
@urso @andrewkroh I wonder how we should move forward here. We made some improvements on the docker and setup side which "mostly" solved the problem but not completely. I'm thinking of also getting this PR in as an "additional" fix? It would need a rebase. WDYT?

File: heartbeat/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM golang:1.8.3
```
Comment (ruflin) [inline heartbeat/Dockerfile]: @urso I assume the heartbeat docker and compose file are not really needed for this PR?

Comment (urso) [reply heartbeat/Dockerfile]: Depending on test environment the tests themselves run in a docker environment. I added the docker container, such that heartbeat test/build environment will be more similar to the other beats.

**Comment by urso** [comment]
+1 for moving forward. I still think these changes are required for a clean test run.

**Comment by ruflin** [comment]
@urso @andrewkroh Going to close this PR for now as CI is working now with fixes on the infra side. We can still come back to this later if we see issues again.

**Pull request closed without merging**